conans/client/loader.py

```diff
@@ -63,7 +63,12 @@ def _load_data(self, conanfile_path):
         if not os.path.exists(data_path):
             return None
 
-        return yaml.safe_load(load(data_path)) or {}
+        try:
+            data = yaml.safe_load(load(data_path))
+        except Exception as e:
+            raise ConanException("Invalid yml format at {}: {}".format(DATA_YML, e))
```

Comment:
We also need to tell the user that this file is automatically loaded by Conan (at this time there is no way to avoid it, the user cannot choose a different name) and maybe also add something like "contents of this file are accessible through self.conan_data"... otherwise, the user will wonder why Conan is worried about this file.

Reply:
But the user put that file there, not me, so he know what is he doing, right? I'm ok with an info trace in the export maybe, but nothing more.

Reply:
The user could already have that file there, and suddenly (with a new version) Conan fails with an error.